src/main/java/com/faforever/client/mod/ModVaultController.java

```diff
@@ -42,167 +31,136 @@
-  private final ReportingService reportingService;
 
-  public Pane searchResultGroup;
-  public Pane searchResultPane;
-  public Pane showroomGroup;
-  public Node loadingLabel;
-  public Pane highestRatedUiPane;
-  public Pane newestPane;
-  public Pane highestRatedPane;
-  public JFXButton moreHighestRatedUiButton;
-  public JFXButton moreNewestButton;
-  public JFXButton moreHighestRatedButton;
-  public StackPane modVaultRoot;
-  public ScrollPane scrollPane;
-  public Button backButton;
-  public SearchController searchController;
-  public Button moreButton;
-
-  private boolean initialized;
+  private ModShowRoomController modShowRoomController;
   private ModDetailController modDetailController;
-  private final ObjectProperty<ModVaultController.State> state;
-  private int currentPage;
-  private Supplier<CompletableFuture<List<ModVersion>>> currentSupplier;
+  private Button manageModButton;
 
   public ModVaultController(ModService modService, I18n i18n, EventBus eventBus, PreferencesService preferencesService,
-                            UiService uiService, NotificationService notificationService, ReportingService reportingService) {
-    this.modService = modService;
-    this.i18n = i18n;
+                               UiService uiService, NotificationService notificationService, ReportingService reportingService) {
+    super(uiService, notificationService, i18n, preferencesService, reportingService);
     this.eventBus = eventBus;
-    this.preferencesService = preferencesService;
-    this.uiService = uiService;
-    this.notificationService = notificationService;
-    this.reportingService = reportingService;
-    state = new SimpleObjectProperty<>(State.LOADING);
+    this.modService = modService;
   }
 
   @Override
   public void initialize() {
     super.initialize();
     JavaFxUtil.fixScrollSpeed(scrollPane);
 
-    loadingLabel.managedProperty().bind(loadingLabel.visibleProperty());
-    showroomGroup.managedProperty().bind(showroomGroup.visibleProperty());
-    searchResultGroup.managedProperty().bind(searchResultGroup.visibleProperty());
-    backButton.managedProperty().bind(backButton.visibleProperty());
-    moreButton.managedProperty().bind(moreButton.visibleProperty());
-
     modDetailController = uiService.loadFxml("theme/vault/mod/mod_detail.fxml");
     Node modDetailRoot = modDetailController.getRoot();
-    modVaultRoot.getChildren().add(modDetailRoot);
+    vaultRoot.getChildren().add(modDetailRoot);
     AnchorPane.setTopAnchor(modDetailRoot, 0d);
     AnchorPane.setRightAnchor(modDetailRoot, 0d);
     AnchorPane.setBottomAnchor(modDetailRoot, 0d);
     AnchorPane.setLeftAnchor(modDetailRoot, 0d);
     modDetailRoot.setVisible(false);
 
+    manageModButton = new Button(i18n.get("modVault.manageMods"));
+    manageModButton.getStyleClass().add("button-raised");
+    manageModButton.setOnAction((event -> manageMods()));
+    searchBox.getChildren().add(manageModButton);
```

Comment:
Avoid defining UI Elements in java code

Reply:
I can add this to the to the vault entity and just hide it for the other classes.

Although with regards to buttons in general we do have to set the onAction in java since the fxml doesn't know about the functions without having a controller set there.